packages/react/src/components/ComposedModal/ComposedModal.js

```diff
@@ -369,29 +370,40 @@ export class ModalHeader extends Component {
   }
 }
 
-export class ModalBody extends Component {
-  static propTypes = {
-    /**
-     * Specify an optional className to be added to the Modal Body node
-     */
-    className: PropTypes.string,
-  };
-
-  render() {
-    const { className, children, ...other } = this.props;
-
-    const contentClass = classNames({
-      [`${prefix}--modal-content`]: true,
-      [className]: className,
-    });
-
-    return (
-      <div className={contentClass} {...other}>
-        {children}
-      </div>
-    );
-  }
+export function ModalBody(props) {
+  const { className, children, hasScrollingContent, ...other } = props;
+  const contentClass = classNames({
+    [`${prefix}--modal-content`]: true,
+    [className]: className,
+  });
+  const hasScrollingContentProps = hasScrollingContent
+    ? {
+        tabIndex: 0,
+        role: 'region',
+      }
+    : {};
+  return (
+    <div className={contentClass} {...hasScrollingContentProps} {...other}>
+      {children}
+    </div>
+  );
 }
+ModalBody.propTypes = {
+  /**
+   * Specify an optional className to be added to the Modal Body node
+   */
+  className: PropTypes.string,
+
+  /**
+   * Specify whether the modal contains scrolling content
+   */
+  hasScrollingContent: PropTypes.bool,
+
+  /**
+   * Required props for the accessibility label of the header
+   */
+  ...AriaLabelPropType,
```

Comment:
Would we consider this a breaking change since we are now requiring `aria-label` or `aria-labelledby` where-as we weren't before?

Reply:
I guess we can consider it a breaking change, does that mean we want to put this behind a feature flag?

Reply:
I'm definitely torn, I feel like we could get away with only supporting `aria-label` and then maybe doing a custom prop type like:

```ts
// Supply the prop type that, when available, means that
// this prop is required
type isRequiredWith = (propName: string, propType: PropType) => PropTypeValidator

// Usage
'aria-label': isRequiredWith('hasScrollingContent', PropTypes.string.isRequired)
```

What do you think? That way this prop is only required if `hasScrollingContent` is true. Unfortunately this might mean that this is off by default so not sure how @dakahn feels about that

Reply:
I think I'd rather go with the custom validator instead of feature flag, similar to https://github.com/carbon-design-system/carbon/blob/master/packages/react/src/prop-types/requiredIfValueExists.js I guess?

Reply:
Yup! Would make sense